File: functions/src/models/prediction-and-score.ts
class PredictionAndScore {
  prediction: number;
  score: number;

  constructor({ prediction, score }: { prediction: number; score: number }) {
    this.prediction = prediction;
    this.score = score;
  }

  toJson() {
    return {
      prediction: this.prediction,
      score: this.score,
    };
  }
}

export { PredictionAndScore };

File: lib/common/models/prediction/prediction_and_score.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'prediction_and_score.freezed.dart';
part 'prediction_and_score.g.dart';

@freezed
class PredictionAndScore with _$PredictionAndScore {
  factory PredictionAndScore({required int prediction, @Default(0) int score}) =
      _PredictionAndScore;

  factory PredictionAndScore.fromJson(Map<String, dynamic> json) =>
      _$PredictionAndScoreFromJson(json);

// static _toJson() {
//   return <String, dynamic>{
//     'prediction': prediction,
//     'score': score,
//   }
// }
}
